Assertion error on CONACK

```
    client = mqtt.Client(client_id=MQTT_CLIENT_ID, protocol=5)
    # Assign event callbacks
    client.on_connect = on_connect
    client.on_publish = on_publish
    client.on_subscribe = on_subscribe
    client.on_log = on_log

    client.on_message = on_message

    client.username_pw_set(MQTT_USER, MQTT_PASS)

    # clean_start=False means this client (with the same client_id)
    # will get queued messages that it hasn't recevied yet
    client.connect(MQTT_HOST, MQTT_PORT, clean_start=False)

    # Continue the network loop, exit when an error occurs
    client.loop_forever()
```

```Python
AssertionError: null
  File "runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "__main__.py", line 4, in <module>
    client.main()
  File "msmqtt/client.py", line 288, in main
    client.loop_forever()
  File "paho/mqtt/client.py", line 1779, in loop_forever
    rc = self.loop(timeout, max_packets)
  File "paho/mqtt/client.py", line 1181, in loop
    rc = self.loop_read(max_packets)
  File "paho/mqtt/client.py", line 1572, in loop_read
    rc = self._packet_read()
  File "paho/mqtt/client.py", line 2310, in _packet_read
    rc = self._packet_handle()
  File "paho/mqtt/client.py", line 2942, in _packet_handle
    return self._handle_connack()
  File "paho/mqtt/client.py", line 2980, in _handle_connack
    reason = ReasonCodes(CONNACK >> 4, identifier=result)
  File "paho/mqtt/reasoncodes.py", line 127, in __init__
    self.getName()  # check it's good
  File "paho/mqtt/reasoncodes.py", line 173, in getName
    return self.__getName__(self.packetType, self.value)
  File "paho/mqtt/reasoncodes.py", line 140, in __getName__
    assert len(namelist) == 1
```

The exception is from this line.
```Python
CONNACK = 0x20
result = 1
from paho.mqtt.client import ReasonCodes
reason = ReasonCodes(CONNACK >> 4, identifier=result)
```

--- a/homeassistant/components/mqtt/manifest.json
+++ b/homeassistant/components/mqtt/manifest.json
@@ -3,7 +3,7 @@
   "name": "MQTT",
   "config_flow": true,
   "documentation": "https://www.home-assistant.io/integrations/mqtt",
-  "requirements": ["paho-mqtt==1.5.1"],
+  "requirements": ["paho-mqtt==1.6.1"],
   "dependencies": ["http"],
   "codeowners": ["@emontnemery"],
   "iot_class": "local_push"

--- a/homeassistant/components/shiftr/manifest.json
+++ b/homeassistant/components/shiftr/manifest.json
@@ -2,7 +2,7 @@
   "domain": "shiftr",
   "name": "shiftr.io",
   "documentation": "https://www.home-assistant.io/integrations/shiftr",
-  "requirements": ["paho-mqtt==1.5.1"],
+  "requirements": ["paho-mqtt==1.6.1"],
   "codeowners": ["@fabaff"],
   "iot_class": "cloud_push"
 }

--- a/homeassistant/package_constraints.txt
+++ b/homeassistant/package_constraints.txt
@@ -19,7 +19,7 @@ home-assistant-frontend==20211103.0
 httpx==0.19.0
 ifaddr==0.1.7
 jinja2==3.0.2
-paho-mqtt==1.5.1
+paho-mqtt==1.6.1
 pillow==8.2.0
 pip>=8.0.3,<20.3
 pyserial==3.5

--- a/requirements_all.txt
+++ b/requirements_all.txt
@@ -1169,7 +1169,7 @@ p1monitor==1.0.0
 
 # homeassistant.components.mqtt
 # homeassistant.components.shiftr
-paho-mqtt==1.5.1
+paho-mqtt==1.6.1
 
 # homeassistant.components.panasonic_bluray
 panacotta==0.1

--- a/requirements_test_all.txt
+++ b/requirements_test_all.txt
@@ -695,7 +695,7 @@ p1monitor==1.0.0
 
 # homeassistant.components.mqtt
 # homeassistant.components.shiftr
-paho-mqtt==1.5.1
+paho-mqtt==1.6.1
 
 # homeassistant.components.panasonic_viera
 panasonic_viera==0.3.6

